recipes/jkqtplotter/meta.yaml

```diff
@@ -0,0 +1,61 @@
+{% set name = "jkqtplotter" %}
+{% set version = "2019.11.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/jkriege2/JKQtPlotter/archive/v{{ version }}.tar.gz
+  sha256: 94d39834c6f4e4afd4ad8511231a26d74fd88dc60d7b3ccd85b7ce00f32ff487
+
+build:
+  number: 0
+  skip: True  # [win and vc<14]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - qt
+    - cmake
+    - make  # [unix]
+    - ninja   # [win]
+    - {{ cdt('mesa-libgl-devel') }}      # [linux]
+    - {{ cdt('mesa-dri-drivers') }}      # [linux]
+    - {{ cdt('mesa-libegl-devel') }}     # [linux]
+    - {{ cdt('libselinux') }}            # [linux]
+    - {{ cdt('libxdamage') }}            # [linux]
+    - {{ cdt('libxfixes') }}             # [linux]
+    - {{ cdt('libxxf86vm') }}            # [linux]
+    - {{ cdt('libxcb') }}                # [linux]
+    - {{ cdt('libxext') }}               # [linux]
+    - {{ cdt('xorg-x11-server-xvfb') }}  # [linux]
+    - {{ cdt('libxau') }}                # [linux]
+    - {{ cdt('libxi-devel') }}           # [linux]
```

Comment:
@isuruf, do you know what these `cdt` lines do? I need the `GL/gl.h` library on Linux, which in Ubuntu is in the `libglu1-mesa-dev` package. I discovered that if I add these, then things build, but I don't know if I need all of them, or what exactly the `cdt` does.

Reply:
Yes, they are needed. The compilers have its own sysroot in `$BUILD_PREFIX/$HOST/sysroot` instead of `/` so that glibc, linux headers from the OS are not used and a package you build in a new Ubuntu version works in an old Ubuntu version. (This works because the compiler will not use the glibc from Ubuntu, but what is in the package which is CentOS 6)
Since the compilers are not looking in `/lib, /usr/lib`, OpenGL is not found and there are no OpenGL conda packages. So, what this does is, download the OpenGL package from CentOS 6 and put it in the sysroot, so that they are used for building.
At runtime though, the user needs to have installed an OpenGL library, but since we are using the one from CentOS6, they'll be compatible. (unless the one installed by user is really really old)

Reply:
I see, thanks for the explanation. Makes sense.